lxd/network/openvswitch/ovs.go

```diff
@@ -373,3 +373,137 @@ func (o *OVS) OVNSouthboundDBRemoteAddress() (string, error) {
+	if err != nil {
+		// default stp-priority is used if stp-priority is not defined
+		if strings.Contains(err.Error(), fmt.Sprintf("no key \"stp-priority\" in Bridge record \"%s\" column other_config", bridgeName)) {
+			return "8000", nil
+		}
+
+		// All other errors are considered errors
+		return "8000", err
+	}
+
+	// Convert to hex format
+	output = strings.TrimSpace(output)
+	// remove surrounding quotes
+	output = output[1 : len(output)-1]
+	stpPriority, err := strconv.Atoi(output)
+	if err != nil {
+		return "8000", err
+	}
+
+	hexStp := fmt.Sprintf("%4X", stpPriority)
+	return hexStp, nil
+}
+
+// GenerateOVSBridgeID returns the bridge ID of the OVS bridge.
+// The bridge IDs follow the following format <STP priority>.<MAC Address>.
+// Check the Bridge ID section on https://www.kernel.org/doc/Documentation/networking/bridge.rst.
+// The STP priority is in hexadecimal format just like the MAC address.
+func (o *OVS) GenerateOVSBridgeID(ctx context.Context, bridgeName string) (string, error) {
+	// get the MAC address
+	netIf, err := net.InterfaceByName(bridgeName)
+	if err != nil {
+		return "None", err
+	}
+
+	bridgeHwID := strings.ReplaceAll(strings.ToLower(netIf.HardwareAddr.String()), ":", "")
+	// Get the STP priority
+	stpPriority, err := GetSTPPriority(ctx, bridgeName)
+	if err != nil {
+		return "None", err
+	}
+
+	return stpPriority + "." + bridgeHwID, nil
+}
+
+// STPEnabled checks if STP is enabled by looking up the "stp_enable" boolean config variable.
+// Returns the value stored in "stp_enable", or false if it is undefined.
+func (o *OVS) STPEnabled(ctx context.Context, bridgeName string) (bool, error) {
+	output, err := shared.RunCommandContext(ctx, "ovs-vsctl", "get", "bridge", bridgeName, "stp_enable")
+	if err != nil {
+		return false, err
+	}
+
+	output = strings.TrimSpace(output)
+	return strconv.ParseBool(output)
+}
+
+// GetSTPForwardDelay returns the STP forward delay in ms. OVS returns the value in seconds, so it needs to be
+// converted to ms to satisfy the api.NetworkStateBridge struct which expects the value in ms.
+// Default value is 15s.
+// Check the "other_config : stp-forward-delay:" section on http://www.openvswitch.org/support/dist-docs/ovs-vswitchd.conf.db.5.txt
+func (o *OVS) GetSTPForwardDelay(ctx context.Context, bridgeName string) (uint64, error) {
+	output, err := shared.RunCommandContext(ctx, "ovs-vsctl", "get", "bridge", bridgeName, "other_config:stp-forward-delay")
+	// get bridge does not differentiate in it's error codes between variables undefined and other errors, such as bridge not found
+	// the error output will need to be checked to make sure if stp-forward-delay is not defined (defaults to 15000) or if a breaking error happened
+	if err != nil {
+		// default stp-priority is used if stp-priority is not defined
+		if strings.Contains(err.Error(), fmt.Sprintf("no key \"stp-forward-delay\" in Bridge record \"%s\" column other_config", bridgeName)) {
+			return 15 * 1000, nil
```

Comment:
Please can you define a constant for this once rather than defining it multiple times.